app.py

```diff
@@ -103,6 +120,66 @@ def simple_flow(times):
 
     my_bulb.start_flow(flow1)
 
+def split_times_to_datetimes(split_times):
+    results = []
+    current_time = datetime.datetime.today()
+    for time in split_times:
+        test_time = str(current_time.year) + " " + str(current_time.month)+ " " + str(current_time.day)+ " " + str(time[0]) + str(time[1])
+        dateobj = datetime.datetime.strptime(test_time, '%Y %m %d %I:%M %p')
+        results.append([dateobj, time[2].split(' ')[0]])
+
+    return results
+
+def get_data_points(current_time, tide_times):
+    results = []
+    prev_time = None
+    next_time = None
+    last_item = None
```

Comment:
If you are going to do stuff like this, it is slightly cleaner to write it: `prev_time = next_time = last_item = None`